src/controllers/GenderCodesController.ts

```diff
@@ -0,0 +1,18 @@
+import { Body, Delete, Get, Path, Post, Put, Query, Route } from 'tsoa';
+import { GenderCode } from '../models/GenderCode';
+import { GenderCodeService } from '../services/GenderCodeService';
+import ControllerBase from './ControllerBase';
+
+@Route('codes/gender')
+export class GenderCodesController extends ControllerBase<GenderCode> {
+
+    get service(){
+        return new GenderCodeService();
+    }
+
+    @Get()
+    public getGenderCodes(){
+        return this.service.getAll();
```

Comment:
Since you inherit from ControllerBase, there are methods provided for you to do this based on the `service` getter.  I suggest taking a quick gander at what is in `ControllerBase` and see how you can use `super.getAll()` here.

The advantage of using base classes is that it become much easier to add / augment functionality of the system as a whole but generally is only valuable if the base class members are used as much as possible.